planner/cascades/implementation_rules.go

```diff
@@ -98,12 +99,14 @@ func (r *ImplTableDual) Match(expr *memo.GroupExpr, prop *property.PhysicalPrope
 }
 
 // OnImplement implements ImplementationRule OnImplement interface.
-func (r *ImplTableDual) OnImplement(expr *memo.GroupExpr, reqProp *property.PhysicalProperty) (memo.Implementation, error) {
+func (r *ImplTableDual) OnImplement(expr *memo.GroupExpr, reqProp *property.PhysicalProperty) ([]memo.Implementation, error) {
 	logicProp := expr.Group.Prop
 	logicDual := expr.ExprNode.(*plannercore.LogicalTableDual)
 	dual := plannercore.PhysicalTableDual{RowCount: logicDual.RowCount}.Init(logicDual.SCtx(), logicProp.Stats, logicDual.SelectBlockOffset())
 	dual.SetSchema(logicProp.Schema)
-	return impl.NewTableDualImpl(dual), nil
+	var duals []memo.Implementation
+	duals = append(duals, impl.NewTableDualImpl(dual))
+	return duals, nil
```

Comment:
This can be simplified as `return []memo.Implementation{impl.NewTableDualImpl(dual), nil}`.